pkg/otlp/attributes/attributes_translator.go

```diff
@@ -0,0 +1,61 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attributes
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes/source"
+)
+
+const missingSourceMetricName string = "datadog.otlp_translator.attributes.missing_source"
+
+// Translator of attributes.
+type Translator struct {
+	missingSources metric.Int64Counter
+}
+
+// NewTranslator returns a new attributes mapper.
+func NewTranslator(set component.TelemetrySettings) (*Translator, error) {
+	meter := set.MeterProvider.Meter("github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes")
+	missingSources, err := meter.Int64Counter(
+		missingSourceMetricName,
+		metric.WithDescription("OTLP payloads that are missing a source (e.g. hostname)"),
+		metric.WithUnit("[payload]"),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build missing source counter: %w", err)
+	}
+
+	return &Translator{
+		missingSources: missingSources,
+	}, nil
+}
+
+// MapToSource gets a telemetry signal source from its attributes.
+func (p *Translator) MapToSource(ctx context.Context, attrs pcommon.Map, set attribute.Set) (source.Source, bool) {
```

Comment:
Seems like the PR is missing the logic to avoid counting `datadog.otlp_translator.attributes.missing_source` ?